docs/reference/indices/downsample-data-stream.asciidoc

```diff
@@ -0,0 +1,157 @@
+[role="xpack"]
+[[indices-downsample-data-stream]]
+=== Downsample index API
+++++
+<titleabbrev>Downsample</titleabbrev>
+++++
+
+preview::[]
+
+Aggregates a time series (TSDS) index and stores
+pre-computed statistical summaries (`min`, `max`, `sum`, `value_count` and
+`avg`) for each metric field grouped by a configured time interval. For example,
+a TSDS index that contains metrics sampled every 10 seconds can be downsampled
+to an hourly index. All documents within an hour interval are summarized and
+stored as a single document in the downsample index.
+
+////
+[source,console]
+----
+PUT /my-time-series-index
+{
+    "settings": {
+        "index": {
+            "mode": "time_series",
+            "time_series": {
+                "start_time": "2022-06-10T00:00:00Z",
+                "end_time": "2022-06-30T23:59:59Z"
+            },
+            "routing_path": [
+                "test.namespace"
+            ],
+            "number_of_replicas": 0,
+            "number_of_shards": 2
+        }
+    },
+    "mappings": {
+        "properties": {
+            "@timestamp": {
+                "type": "date"
+            },
+            "metric": {
+                "type": "long",
+                "time_series_metric": "gauge"
+            },
+            "dimension": {
+                "type": "keyword",
+                "time_series_dimension": true
+            }
+        }
+    }
+}
+
+PUT /my-time-series-index/_block/write
+
+----
+// TEST
+////
+
+[source,console]
+----
+POST /my-time-series-index/_downsample/my-downsampled-time-series-index
+{
+    "fixed_interval": "1d"
+}
+----
+// TEST[continued]
+
+////
+[source,console]
+----
+DELETE /my-time-series-index*
+DELETE _data_stream/*
+DELETE _index_template/*
+----
+// TEST[continued]
+////
+
+[[downsample-api-request]]
+==== {api-request-title}
+
+`POST /<source-index>/_downsample/<output-downsampled-index>`
+
+[[downsample-api-prereqs]]
+==== {api-prereq-title}
+
+* Only indices in a <<tsds,time series data stream>> are supported.
+
```

Comment:
When it comes to permissions requirements are the following:
* `all` or `manage` are required (those for instance, as satisfied by an admin user normally)
* no Field Level Security and no Document Level Secuirty should be defined on the source index. We check if FLS or DLS is defined on the source idnex and we fail downsampling if that is the case.

Reply:
Thanks for chiming in, Salvatore

Reply:
Thanks Salvatore!
The prereqs are now:

* Only indices in a <<tsds,time series data stream>> are supported.

* If the {es} {security-features} are enabled, you must have the `all`
or `manage` <<privileges-list-indices,index privilege>> for the data stream.

* Neither <<field-and-document-access-control,field nor document level security>> can be defined on the source index.

* The source index must be read only (`index.blocks.write: true`).